geode-core/src/main/java/org/apache/geode/internal/cache/BucketAdvisor.java

```diff
@@ -896,9 +888,10 @@ private void removePrimary(InternalDistributedMember member) {
           ((BucketRegion) br).beforeReleasingPrimaryLockDuringDemotion();
         }
 
-        releasePrimaryLock();
         // this was a deposePrimary call so we need to depose children as well
         deposePrimaryForColocatedChildren();
+        releasePrimaryLock();
```

Comment:
Do we need to make this change? This is releasing the dlock after all children are deposed as primary. I thought just the fact that we are holding the primary move write lock this whole time will prevent any operations from being in progress on any child/grandchild region while this depose is going on.

Reply:
releasePrimaryLock() will release the distributed lock on parent region. Now any node (including the one trying to become primary) can get that lock and becomes primary. And operation can be generated on the new primary on parent region. This change will guarantee that all colocated buckets deposed primary before another server can become new primary for any colocated buckets.

You are also right that node with original primary colocated buckets should be blocked on operations on the colocated regions. Even though new primary is being acquired by another server, the original old primary will not allow any new operations on the colocated child buckets. I guess only one is needed.